File: tests/sdk/nodejs/test/sts.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import { GetCallerIdentityCommand } from "@aws-sdk/client-sts";
import { newSTS, requireReady } from "../lib/helpers.mjs";

test("STS GetCallerIdentity", async (t) => {
  if (!(await requireReady(t))) return;
  const client = newSTS();
  const out = await client.send(new GetCallerIdentityCommand({}));
  assert.ok(out.Account, "expected Account");
  assert.ok(out.Arn, "expected Arn");
  assert.ok(out.UserId, "expected UserId");
});

File: tests/sdk/go/sts_test.go
package sdk_test

import (
	"context"
	"testing"

	"github.com/aws/aws-sdk-go-v2/service/sts"
)

func TestSTSGetCallerIdentity(t *testing.T) {
	requireReady(t)
	cfg := loadAWSConfig(t)
	client := newSTS(t, cfg)

	out, err := client.GetCallerIdentity(context.Background(), &sts.GetCallerIdentityInput{})
	if err != nil {
		t.Fatalf("GetCallerIdentity: %v", err)
	}
	if out.Account == nil || *out.Account == "" {
		t.Fatal("expected Account")
	}
	if out.Arn == nil || *out.Arn == "" {
		t.Fatal("expected Arn")
	}
	if out.UserId == nil || *out.UserId == "" {
		t.Fatal("expected UserId")
	}
	t.Logf("caller account=%s arn=%s", *out.Account, *out.Arn)
}
